postgres_deparse.h

```diff
@@ -0,0 +1,34 @@
+#ifndef POSTGRES_DEPARSE_H
+#define POSTGRES_DEPARSE_H
+
+#include <stdbool.h>
+#include <sys/types.h>
+
+typedef struct PostgresDeparseComment {
+    int match_location;          // Insert comment before a node, once we find a node whose location field is equal-or-higher than this location
+    int newlines_before_comment; // Insert newlines before inserting the comment (set to non-zero if the source comment was separated from the prior token by at least one newline)
+    int newlines_after_comment;  // Insert newlines after inserting the comment (set to non-zero if the source comment was separated from the next token by at least one newline)
+    char *str;                   // The actual comment string, including comment start/end tokens, and newline characters in comment (if any)
+} PostgresDeparseComment;
+
+typedef struct PostgresDeparseOpts {
+    PostgresDeparseComment **comments;
+    size_t comment_count;
+
+    // Pretty print options
+    bool pretty_print;
+    int indent_size;           // Indentation size (Default 4 spaces)
```

Comment:
I still think 2 spaces would generally work better for this kind of formatting, but since it's configurable, it's not a big deal.